### PR TITLE
Hermit PSBT signing for pre-product key recovery

### PR DESCRIPTION
**Pull request by humanumbrella**
## Description

This PR simply attempts to bring an existing feature branch into the mainline -- it is working in the feature branch, but we don't want to leave the code dangling there indefinitely. This functionality is used for pre-product key recovery signing via hermit and is, therefore, a bit challenging to test. I have a write up somewhere, but want to get this branch up to date with `master` and more or less ready to go. My goal is to put this code in to `master` ... and have it be the first attempt next time it's needed.

## Does this PR introduce a breaking change?

* [x] Yes
* [ ] No

It removes pubkey exporting from hermit, which is no longer supported.

## Does this PR fix an open issue?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/actions/transactionActions.js [hunk 1]
```diff
@@ -277,3 +295,127 @@ export function setMaxSpendOnOutput(outputIndex) {
...
+export function importLegacyPSBT(psbtText) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const { network } = state.settings;
+    const psbt = autoLoadPSBT(psbtText, { network: networkData(network) });
+    if (!psbt) {
+      throw new Error("Could not parse PSBT.");
+    }
+
+    // FIXME make sure all redeem scripts are the same -- can only support this kind of spend
```
Comment (bucko13) [inline src/actions/transactionActions.js:415]: Cleanup? TODO? Log as an issue?

File: src/actions/transactionActions.js [hunk 2]
```diff
@@ -277,3 +295,127 @@ export function setMaxSpendOnOutput(outputIndex) {
...
+    if (psbt.txInputs.length === 0) {
+      throw new Error("PSBT does not contain any inputs.");
+    }
+    if (psbt.txOutputs.length === 0) {
+      throw new Error("PSBT does not contain any outputs.");
+    }
+
+    dispatch(resetOutputs());
+    dispatch(setUnsignedPSBT(psbt.toBase64()));
+    // now do sigs
```
Comment (bucko13) [inline src/actions/transactionActions.js:402]: What does this comment mean? Can we remove it now?

File: src/components/Hermit/HermitSignatureImporter.jsx [hunk 3]
```diff
@@ -1,47 +1,159 @@
...
       outputs,
-      bip32Paths,
+      setUnsignedPSBT,
+      unsignedPsbtFromState,
+    } = this.props;
+    let psbtToSign;
+    // We need to be flexible here because this signature importer is used in multiple places
+    // And the user *could* have uploaded their own PSBT, and that uploaded PSBT *could* also
+    // be a scaffolded PSBT without any inputs.
+    //
```
Comment (bucko13) [inline src/components/Hermit/HermitSignatureImporter.jsx:70]: nit cleanup

File: src/components/ScriptExplorer/ScriptEntry.jsx [hunk 4]
```diff
@@ -259,11 +277,90 @@ class ScriptEntry extends React.Component {
...
+          //   // setNetwork(psbt.opts.network); -- BUGFIX needed on buidl psbt builder
+          //   // setAddress(1, output.address);
+          //   setAmount(1, satoshisToBitcoins(output.value).toFixed(8));
+          // setAddress(1, "2MuAeixpNw4xWMKqvhz8YEVEodWXRxCYsXG");
+          //   const inputsTotalSats = new BigNumber(10000000);
+          //   const feeSats = inputsTotalSats - outputsTotalSats;
+          //   const fee = satoshisToBitcoins(feeSats).toFixed(8).toString();
+          //   setFee(fee);
+          //   finalizeOutputs(true);
+          // });
```
Comment (bucko13) [inline src/components/ScriptExplorer/ScriptEntry.jsx:342]: can this be cleaned up?

File: src/reducers/transactionReducer.js [hunk 5]
```diff
@@ -399,6 +402,8 @@ export default (state = initialState(), action) => {
       return validateTransaction(updateOutputAddress(state, action));
     case SET_OUTPUT_AMOUNT:
       return validateTransaction(updateOutputAmount(state, action));
+    case SET_UNSIGNED_PSBT:
+      return updateState(state, { unsignedPSBT: action.value });
```
Comment (bucko13) [inline src/reducers/transactionReducer.js:406]: future TODO but I suspect this probably didn't need to be in the redux store at all. Probably something we need to start moving away from eventually in Caravan.

**Review by bucko13 — COMMENTED** [review]
Mostly comments about cleanup.

**Review by bucko13 — CHANGES_REQUESTED** [review]
One tiny request on a leftover comment that needs some more clarification otherwise I think this is good enough to merge. 

It would be nice to refactor some of this in the future, simplifying the actions or removing them altogether as necessary. After some of our work on PSBTv2 in unchained-bitcoin I think there's particularly room to improve the API of managing the PSBT state at the application layer. to make these components a little easier to manage/maintain and understand what's going on. 

All for future work though!

File: src/actions/transactionActions.js [hunk 6]
```diff
@@ -277,3 +295,128 @@ export function setMaxSpendOnOutput(outputIndex) {
...
+
+    state = getState();
+    const inputsTotalSats = BigNumber(state.spend.transaction.inputsTotalSats);
+    const feeSats = inputsTotalSats - outputsTotalSats;
+    const fee = satoshisToBitcoins(feeSats).toFixed(8);
+    dispatch(setFee(fee));
+
+    dispatch(finalizeOutputs(true));
+
+    // now do sigs
```
[the remark this answers is not in the document]
Comment (humanumbrella) [reply src/actions/transactionActions.js:380]: i thought i deleted that comment - if we want to support loading a **partially signed** PSBT ... we need to pay attention to included signatures in the psbt. i chose not to support that initially. will remove this comment.

**Review by bucko13 — APPROVED** [review]

**Pull request merged**